instruments/abstract_instruments/comm/abstract_comm.py

```diff
@@ -202,7 +203,14 @@ def read(self, size=-1, encoding="utf-8"):
         :return: The read string from the connection
         :rtype: `str`
         """
-        return self.read_raw(size).decode(encoding)
+        if encoding == 'utf-8':
+            return self.read_raw(size).decode(encoding)
+
+        elif encoding == 'IEEE-754/64':
+            return struct.unpack('>d', self.read_raw(size))[0]
+
+        else:
+            raise NotImplementedError
```

Comment:
We should probably include a little blurb in the raised exception for the error.

Actually, should this be `NotImplementedError` or should it be `ValueError` ? I can see how other encodings might be valid just not implemented yet. Either way, something like this would be good:

```python
raise NotImplementedError("Encoding {} is not currently supported".format(encoding))
```

Reply:
You are right,  [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) makes more sense. I'll fix that.